reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/VariableDeclToExprReductionOpportunities.java

```diff
@@ -53,4 +55,28 @@ public VariableDeclToExprReductionOpportunities(TranslationUnit tu, ReducerConte
     finder.visit(tu);
     return finder.getOpportunities();
   }
+
+  @Override
+  public void visitDeclarationStmt(DeclarationStmt declarationStmt) {
+    super.visitDeclarationStmt(declarationStmt);
+    if (!context.reduceEverywhere()) {
+      // Replacing variable declaration with a new binary expression might have a side-effect,
+      // do not consider these reduction opportunities if we are not reducing everywhere.
+      return;
+    }
+    final List<VariableDeclInfo> declInfos =
+        declarationStmt.getVariablesDeclaration().getDeclInfos();
+    // We iterate backwards so that when applying reduction the new expression is inserted at the
+    // correct order with respect to its original order in the variable declaration info list.
+    for (int i = declInfos.size() - 1; i >= 0; i--) {
+      final VariableDeclInfo variableDeclInfo = declInfos.get(i);
```

Comment:
Also check that the initializer is a ScalarInitializer (as that's a condition we require later).

It might be that we get rid of the notion of ScalarInitializer due to #520, but still we should assume for now that there are multiple kinds of initializer and that we are only supporting one.

Reply:
Fixed